src/main/java/functions/VisitorFunction.kt

```diff
@@ -1,5 +1,35 @@
 package functions
 
+import ast.ASTNode
+import ast.Child
+import ast.Node
+import ast.Sibling
 import tree.TargetTreeNode
 
 typealias VisitorFunction = (TargetTreeNode) -> List<TargetTreeNode>
+
+fun createVisitorFunction(node: ASTNode): VisitorFunction {
+    return when (node) {
+        is Child -> {
+            childFunction(
+                createVisitorFunction(node.father),
+                createVisitorFunction(node.child)
+            )
+        }
+        is Node -> {
+            nodeFunction(
+                node.name
+            )
+        }
+        is Sibling -> {
+            val siblingArray = listOf<VisitorFunction>().toMutableList()
+            node.siblings.forEach {
+                siblingArray.add(createVisitorFunction(it))
+            }
+            siblingFunction(siblingArray)
+        }
+        else -> {
+            TODO("Return value not defined")
+        }
+    }
+}
```

Comment:
I believe that we can have a functional approach and do something like this:

```suggestion

fun createVisitorFunction(node: ASTNode): VisitorFunction = when (node) {
        is Child -> 
            childFunction(createVisitorFunction(node.father), createVisitorFunction(node.child))
        is Node -> nodeFunction( node.name)
        is Sibling -> siblingFunction(node.siblings.map {createVisitorFunction(it)})
        else -> throw IllegalArgumentException("This ASTNode subtype is not supported.")
    }
}
```
Do you think this might work? 😄 
I don't know if the `lint` will complain if we don't use `{` `}` blocks. 
![](https://media.giphy.com/media/119LVmecQWrzlm/giphy.gif)